lib/Bio/EnsEMBL/DataCheck/Checks/DescriptionDisplayXrefs.pm

```diff
@@ -0,0 +1,77 @@
+=head1 LICENSE
+
+Copyright [2018-2020] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::DescriptionDisplayXrefs;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'DescriptionDisplayXrefs',
+  DESCRIPTION    => '
+                     Tests for {ECO: } blocks from Uniprot in descriptions.
+                     Test for Bad characters in display_label
+                     Tests for improper import of MIM data.         
+                    ',
+  GROUPS         => ['core', 'xref'],
+  DATACHECK_TYPE => 'critical',
+  TABLES         => ['xref'],
+  PER_DB         => 1
+};
+
+sub tests {
+  my ($self) = @_;
+
+  my $desc_1 = 'No {ECO: } blocks from Uniprot in descriptions';
+  my $sql_1  = qq/
+      SELECT count(*) FROM xref  
+      WHERE description like '%{ECO:%}%'
+  /;
+
+  is_rows_zero($self->dba, $sql_1, $desc_1);
+
+  my $desc_2 = 'No NULL OR Bad Chracters Found In display_label';
+  my $sql_2  = qq/
+      SELECT count(*) FROM xref 
+      WHERE display_label IS NULL
+      OR display_label REGEXP '^[:;\\n\\r\\t~ ]+\$'
```

Comment:
This regexp doesn't work, for a couple of reasons. First, the double-escaping of n, r and t means that it's looking for those literal characters, rather than whitespace - one escape is all that's needed. Also it's looking for text which consists entirely of these characters, but that doesn't seem sensible - we want text that contains them. But then, I think that would actually be too strict for a critical test - we don't have control over the labels that external sources use, and while some of these seem ill-advised, we may have to tolerate them. The human db has examples of spaces, : and ;
```suggestion
      OR display_label REGEXP '[\n\r\t]+'
```

Reply:
We could create an advisory datacheck to check for the characters that I suggest removing, but I don't think it's worth it - as these are externally controlled, we can't do much in the event of a failure.